src/flamenco/runtime/program/fd_stake_program.c

```diff
@@ -905,54 +905,80 @@ stake_weighted_credits_observed( fd_stake_t const * stake,
                                  ulong              absorbed_lamports,
                                  ulong              absorbed_credits_observed,
                                  ulong *            out ) {
+  /* https://github.com/anza-xyz/agave/blob/dc74c22960b4f2adfc672f6dc3bfaa74ec1d5d48/programs/stake/src/stake_state.rs#L1194 */
+  /* if stake.credits_observed == absorbed_credits_observed { */
+  /*     Some(stake.credits_observed) */
   if( FD_LIKELY( stake->credits_observed == absorbed_credits_observed ) ) {
     *out = stake->credits_observed;
     return 1;
   } else {
-    /* total_stake = stake->delegation.stake + absorbed_lamports */
-    ulong total_stake_h = 0UL;
-    ulong total_stake_l = 0UL;
-    fd_uwide_inc( &total_stake_h, &total_stake_l, 0UL, stake->delegation.stake,
-                  absorbed_lamports );
+    /* https://github.com/anza-xyz/agave/blob/dc74c22960b4f2adfc672f6dc3bfaa74ec1d5d48/programs/stake/src/stake_state.rs#L1197 */
+    /* let total_stake = u128::from(stake.delegation.stake.checked_add(absorbed_lamports)?); */
+    ulong total_stake = stake->delegation.stake + absorbed_lamports;
+    /* If there is an overflow on the ulong addition then exit */
+    if( FD_UNLIKELY( stake->delegation.stake < total_stake ) ) {
```

Comment:
Was this PR solving a fuzz mismatch? If so could we also convert the fuzz mismatch into a test fixture and place it in https://github.com/firedancer-io/test-vectors/tree/main/instr/fixtures/stake ? Thanks so much :)